### PR TITLE
[download_dsyms] Added 'live' version parameter to only download live_version dSYMs

### LINKED ISSUE #15308: `download_dsyms` action is missing a fetching 'live' version option
#### Motivation Behind Feature
We have setup a Jenkins job that executes every 3 days to download the dSYMs from App Store Connect and uploads it to New Relic.

While doing that, we noticed that `download_dsyms` only has the option of `version: 'lastest'` or passing specific build version.

But since we are disabling BITCODE_ENABLED for our tests builds (faster waiting for processing time), we would only need to upload dSYM files for our latest released App Store live version.

As our apps are always in heavy development, 'lastest' option doesn't return our desired version.

Example app versions in App Store Connect:

* **3.9.1** (live_version, released App Store version)
* **4.0** (edit_version, waiting for review)
* **4.1** (test build uploaded to TestFlight only)

`'latest'` would return a build from the **4.1** train.

#### Feature Description

Add another `version: 'live'` option to the action to only fetch the latest released App Store live version.

I'm currently preparing a PR for this (my first one ^^)

### PR DESCRIPTION
**Pull request by Buju77**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
This should fix https://github.com/fastlane/fastlane/issues/15308

### Description
Added a new `if version == 'live'` next to `if version == 'latest'` check to only fetch the dSYMs of latest live version.

### Testing Steps
Added a new test in `fastlane/spec/actions_specs/download_dsyms_spec.rb`

### CODE REVIEW

File: fastlane/spec/actions_specs/download_dsyms_spec.rb
Comment (Buju77) [inline fastlane/spec/actions_specs/download_dsyms_spec.rb]: full disclosure: I was not sure how to adapt the tests with a live_version. I hope this is correct 😅

**Comment by Buju77** [comment]
@googlebot I signed it!

**Comment by Buju77** [comment]
@googlebot I fixed it.

File: fastlane/lib/fastlane/actions/download_dsyms.rb
Comment (janpio) [inline fastlane/lib/fastlane/actions/download_dsyms.rb]:
```suggestion
                                       description: "The app version for dSYMs you wish to download, pass in 'latest' to download only the latest build's dSYMs or 'live' to download only the live version dSYMs",
```

File: fastlane/lib/fastlane/actions/download_dsyms.rb
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/download_dsyms.rb]: Should this be `if latest_live_version.nil?` I _think_ if there is no live version then `latest_live_version` will be nil 🤔

```suggestion
          UI.user_error!("Could not find live version for your app, please try setting 'latest' or a specific version") if latest_live_version.nil?
```

File: fastlane/lib/fastlane/actions/download_dsyms.rb
Comment (joshdholtz) [inline fastlane/lib/fastlane/actions/download_dsyms.rb]: Can we rename this to just `live_version`?

File: fastlane/spec/actions_specs/download_dsyms_spec.rb
Comment (joshdholtz) [inline fastlane/spec/actions_specs/download_dsyms_spec.rb]: I think the `allows` from here and below can be deleted since we really only care about `app.live_version`, right?

**Review by joshdholtz — CHANGES_REQUESTED** [review]
This looks good! Just a few small questions/comments and then this should be good to merge after that 😊

File: fastlane/lib/fastlane/actions/download_dsyms.rb
Comment (Buju77) [reply fastlane/lib/fastlane/actions/download_dsyms.rb]: ah true, good catch thx 😅

File: fastlane/spec/actions_specs/download_dsyms_spec.rb
Comment (Buju77) [reply fastlane/spec/actions_specs/download_dsyms_spec.rb]: I'm new to this test setup, so I wasn't sure if I should leave it or not. Because my PR was to fix getting the live version, if there is already an edit_version *and* even a newer build train, which was the initial issue we discovered. should I remove it or keep it?

**Comment by joshdholtz** [comment]
@googlebot I consent.

File: fastlane/spec/actions_specs/download_dsyms_spec.rb
Comment (joshdholtz) [reply fastlane/spec/actions_specs/download_dsyms_spec.rb]: I added a commit to remove this for you! It can be removed because the latest code is being executed in a different if statement during the test so nothing needs to get mocked up for that 😊 After the tests run I will get this merged in. Thanks for contribution ❤️

**Review by joshdholtz — APPROVED** [review]
Thanks for the contribution ❤️

File: fastlane/spec/actions_specs/download_dsyms_spec.rb
Comment (Buju77) [reply fastlane/spec/actions_specs/download_dsyms_spec.rb]: cool thx! 🎉

**Pull request merged**
